src/puppetlabs/trapperkeeper/services/status/check.clj

```diff
@@ -0,0 +1,26 @@
+(ns puppetlabs.trapperkeeper.services.status.check
+  "Shared status check functions."
+  (:require [clojure.java.jdbc :as sql])
+  (:import [java.io File IOException]))
+
+(defn db-up?
+  [db-spec]
+  (try (let [select-42 "SELECT (a - b) AS answer FROM (VALUES ((7 * 7), 7)) AS x(a, b)"
+             [{:keys [answer]}] (sql/query db-spec [select-42])]
+         (= answer 42))
+    (catch Exception _
+      false)))
+
+(defn disk-writable?
+  []
+  (let [temp-file (File/createTempFile "tk-disk-check" ".txt")
```

Comment:
yeah, java docs say it will throw "IOException - If a file could not be created"